cells/lib/ops/mkOperableScript.nix

```diff
@@ -18,13 +17,34 @@ in
   */
   {
     package,
-    args ? {},
+    args ? [],
     bin ? "",
   }: let
-    # Cumulatively builds up arguments: args+=(\"name\" \"value\")\n....
-    args' = builtins.concatStringsSep "\n" (
-      l.mapAttrsToList (name: value: "args+=(\"${name}\" \"${value}\")") args
-    );
+    # Cumulatively builds up arguments...
+    args' = let
+      attrsToStr = {
+        set,
+        prefix ? "",
+        suffix ? "",
+      }:
+        builtins.concatStringsSep "\n" (
+          l.mapAttrsToList (name: value: ''${prefix}"${name}" "${toString value}"${suffix}'') set
+        );
+    in
+      if builtins.isAttrs args
+      then
+        attrsToStr {
```

Comment:
Can you incorporate a deprecation message in `deprecations.nix` and invoke it for this code path?

Reply:
ok, not needed, it's still a hoisting simplification of the singleton `[{}]`.